### DISCUSSION TITLE
Helm install without root privilege on containers.

### DISCUSSION THREAD

**Discussion by mat-ej** [PrefectHQ/prefect#4527] [Q&A]
Hello PrefectHQ team,
your tool looks amazing when running on localhost. I am currently in the process of deploying it to a remote kubernetes cluster where containers cannot be run with root. 

What I usually for helm installs is add the following securityContext for services:

```
securityContext:
    runAsUser: 1000
    runAsGroup: 1000
```

However I am unable to do so with your helm chart:
[https://github.com/PrefectHQ/server/blob/master/helm/prefect-server](https://github.com/PrefectHQ/server/blob/master/helm/prefect-server)

The error I get is the following:
`Error: YAML parse error on prefect-server/templates/hasura/deployment.yaml: error converting YAML to JSON: yaml: line 31: mapping values are not allowed in this context`

Is there a way to deploy prefect-server on kubernetes cluster without root privilege for containers? (I.e. container has runAsNonRoot)

Could you please help me somehow?

Thanks and have great day. :)

**Comment by yasinzaehringer-paradime** [comment]
I am not associated with prefect, but I just had a look at line 31 of this file, i.e. [here](https://github.com/PrefectHQ/server/blob/master/helm/prefect-server/templates/hasura/deployment.yaml#L28-L31). Are sure this is related to the `securityContext` and not related to the `podSecurityContext`? In general it looks to me - see [here](https://stackoverflow.com/questions/44327773/mapping-values-are-not-allowed-in-this-context) - that these errors are YAML errors and are not necessarily related the template itself.
